utils/eps_to_csv_converter.py

```diff
@@ -0,0 +1,428 @@
+        logger.debug(
+            f"Generating ID token for SA '{target_service_account_email}' with audience '{audience}'"
+        )
+        client = iam_credentials_v1.IAMCredentialsClient()
+        name = f"projects/-/serviceAccounts/{target_service_account_email}"
+        id_token_response = client.generate_id_token(
+            name=name, audience=audience, include_email=True
+        )
+        id_token_jwt = id_token_response.token
+        logger.debug("Creating authorized session with generated ID token.")
+        iap_creds = google.oauth2.credentials.Credentials(id_token_jwt)
+        authed_session = AuthorizedSession(iap_creds)
+
+        logger.info(f"Making {method} request to IAP URL: {url}")
+
+        resp = authed_session.request(method, url, **kwargs)
+        logger.info(f"Received status code: {resp.status_code} from {url}")
+        resp.raise_for_status()  # Raises HTTPError for bad responses (4xx or 5xx)
+        try:
+            return resp.json()
+        except JSONDecodeError as json_err:
+            logger.error(
+                f"Failed to decode JSON response from {url}. Response text: {resp.text[:500]}..."
+            )  # Log part of the response
+            raise ValueError(f"Invalid JSON response received from {url}") from json_err
+    except (GoogleAuthError, RequestException) as e:
+        logger.error(f"Error making IAP request to {url}: {e}", exc_info=True)
+        raise e
+    except Exception as e:
+        logger.exception(f"An unexpected error occurred during IAP request to {url}")
+        # Re-raise or handle as appropriate for the application context
+        raise e
+
+
+def remove_prefix_any(text: str, prefixes: List[str]) -> str:
+    """Removes the first matching prefix from a string.
+
+    Args:
+        text: The input string.
+        prefixes: A list of prefixes to check for.
+
+    Returns:
+        The string with the first matching prefix removed, or the original string.
+    """
+    for prefix in prefixes:
+        if text.startswith(prefix):
+            return text[len(prefix):]
+    return text
+
+
+def process_data(
+        data: Dict, prefixes_to_remove: List[str], rename_rules: Dict[str, str]
+) -> pd.DataFrame:
+    """Flattens JSON data, removes prefixes, handles duplicated column names scenarios, validates rename_rules and
+    renames columns.
+
+    Args:
+        data: The raw data dictionary (expected to have a 'clusters' key).
+        prefixes_to_remove: List of prefixes to remove from column names.
+        rename_rules: Specific column renaming rules (original_name: new_name).
+
+    Returns:
+        A processed Pandas DataFrame
+
+    Raises:
+        ValueError: If the input data is invalid.
+        Exception: For unexpected errors during processing.
+    """
+    if "clusters" not in data:
+        raise ValueError("Input data is missing the required 'clusters' key.")
+
+    try:
+        flattened_df = pd.json_normalize(data["clusters"], sep="_")
```

Comment:
ahhh okay so data and intent will be flattened such that they are prefixed accordingly?  So the cluster object has a nested data, so all data keys are flattened to `data_foo`, `data_bar`; intent is flattened to `intent_foo`, `intent_bar`?